python/ray/dag/compiled_dag_node.py

```diff
@@ -173,6 +174,106 @@ def do_profile_tasks(
         raise
 
 
+@DeveloperAPI
+def do_stream_tasks(
+    self,
+    tasks: List["ExecutableTask"],
+    schedule: List[_DAGNodeOperation],
+) -> None:
+    """A generic actor method to begin executing the operations belonging to an
+    actor in a streamed fashion (i.e., allowing overlapping computation and
+    communication). This runs an infinite loop to execute each _DAGNodeOperation
+    in the order specified by the schedule, with dedicated read, compute and write
+    Cuda streams. It exits only if the actor dies or an exception is thrown.
+
+    Args:
+        tasks: the executable tasks corresponding to the actor methods.
+        schedule: A list of _DAGNodeOperation that should be executed in order.
+    """
+    try:
+        import cupy as cp
+        import torch
+        from ray.air._internal import torch_utils
+
+        device = torch_utils.get_devices()[0]
+        exec_stream = cp.cuda.ExternalStream(
+            torch.cuda.Stream().cuda_stream, device_id=device.index
+        )
+
+        for task in tasks:
+            task.prepare()
+
+        done = False
+        while True:
+            if done:
+                break
+            for operation in schedule:
+                done = tasks[operation.exec_task_idx].stream_operation(
+                    self, operation, exec_stream
+                )
+                if done:
+                    break
+    except Exception:
+        logging.exception("Compiled DAG task exited with exception")
+        raise
+
+
+@DeveloperAPI
+def do_profile_stream_tasks(
+    self,
+    tasks: List["ExecutableTask"],
+    schedule: List[_DAGNodeOperation],
+) -> None:
+    """Similar to `do_stream_tasks`, but with torch profiling enabled.
```

Comment:
Suggest starting/stopping the profiling context in a Ray task before/after sending the tasks to start the execution loop. That way we can reuse the current `do_exec_tasks` codepath.